components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -399,14 +401,28 @@ func (g *rtkSerial) connectAndParseSourceTable() error {
 
 	g.urlMutex.Lock()
 	defer g.urlMutex.Unlock()
+
 	err := g.connect(g.ntripClient.URL, g.ntripClient.Username, g.ntripClient.Password, g.ntripClient.MaxConnectAttempts)
 	if err != nil {
 		g.err.Set(err)
 		return g.err.Get()
 	}
 
 	if !g.ntripClient.Client.IsCasterAlive() {
-		g.logger.Infof("caster %s seems to be down", g.ntripClient.URL)
+		g.logger.Infof("caster %s seems to be down, retrying", g.ntripClient.URL)
+		attempts := 0
+		// we will try to connect to the caster five times if it's down.
+		for attempts < 5 {
+			if !g.ntripClient.Client.IsCasterAlive() {
```

Comment:
if i understand correctly this is checking if the caster is alive 5 times instead of trying to reconnect 5 times. Would adding a short sleep after the connect call before checking if the caster is alive solve the issue? If not maybe just change the comment to be more clear.

Reply:
`IsCasterAlive` calls `GetSourceTable` which actually tries to connect to the caster via TCP connection and depending on if the sourcetable is obtained or not `IsCasterAlive` returns `true` or `false`. The naming of these functions are a little weird but it is trying to reconnect.

Reply:
ahhh okay thanks for explaining!